lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -383,16 +383,16 @@ function isObjAuthorized(bucket, objectMD, requestTypesInput, canonicalID, authI
             ? _requestType.slice(0, -7) : _requestType;
         const bucketOwner = bucket.getOwner();
         if (!objectMD) {
-        // User is already authorized on the bucket for FULL_CONTROL or WRITE or
-        // bucket has canned ACL public-read-write
-            if (parsedMethodName === 'objectPut' || parsedMethodName === 'objectDelete') {
-                results[_requestType] = actionImplicitDenies[_requestType] === false;
-                return results[_requestType];
-            }
             // check bucket has read access
             // 'bucketGet' covers listObjects and listMultipartUploads, bucket read actions
             results[_requestType] = isBucketAuthorized(bucket, 'bucketGet', canonicalID, authInfo, log, request,
                 actionImplicitDenies);
+            // User is already authorized on the bucket for FULL_CONTROL or WRITE or
+            // bucket has canned ACL public-read-write
+            if ((parsedMethodName === 'objectPut' || parsedMethodName === 'objectDelete')
+                && results[_requestType] === false) {
+                results[_requestType] = actionImplicitDenies[_requestType] === false;
+            }
```

Comment:
As an optional suggestion, it might make the logic easier to follow written this way I think, so we understand that a positive "bucket authorized" result trumps any object ACL result:

```suggestion
            if (results[_requestType] !== false) {
                // or `return true`?
                return results[_requestType];
            }
            // User is already authorized on the bucket for FULL_CONTROL or WRITE or
            // bucket has canned ACL public-read-write
            if (parsedMethodName === 'objectPut' || parsedMethodName === 'objectDelete') {
                results[_requestType] = (actionImplicitDenies[_requestType] === false);
            }
            return results[_requestType];
```

Also I don't see where saving the results in the `results` object is needed, if not used it could simplify the code, but haven't looked very thoroughly and it's not related to the PR.

Reply:
Thanks for the suggestion, I'll keep it as is as it makes the code less lengthy. The results object is needed in the different checks throughout the code.